### PR TITLE
ci/ipsec: Cilium agents in ci-ipsec-e2e no longer share host's boot ID

### PR DESCRIPTION
**Pull request by jschwinger233**
Containers on the same host share boot id (/proc/sys/kernel/random/boot_id), previous IPsec kind-based e2e tests didn't fully cover the per-node-pair key feature.

This PR introduces a new hidden flag `boot-id-file` for specifying boot ID path, allows us to cover the missing scenario in CI.

ci-ipsec-upgrade can't be covered in this PR because 1.16 still shares boot ID. Upgrade from 1.16 to 1.17 in CI will break connectivity as new cilium suddenly uses a different boot ID.

### CODE REVIEW

**Comment by jschwinger233** [comment]
If I can get approvals from reviewers, I'll drop the last commit [ci: Temporarily change pull_request_target to pull_request](https://github.com/cilium/cilium/pull/35951/commits/67725afa6f1c76d5b0aa2e70c7b9d6386294cb07) before merge. 

Add `don't-merge` label to remind myself.

File: .github/workflows/conformance-ipsec-e2e.yaml
Comment (pchaigno) [inline .github/workflows/conformance-ipsec-e2e.yaml]: Don't we need the same for the upgrade workflow?

**Review by pchaigno — APPROVED** [review]
I don't mind this approach, but I'm wondering why you went this way rather than adding a hidden agent flag? AFAICS, there's no risk of silently breaking security here. If some user manages to use the hidden flag, they'll just break connectivity, as is already possible with many other flags.

File: .github/workflows/conformance-ipsec-e2e.yaml
Comment (jschwinger233) [reply .github/workflows/conformance-ipsec-e2e.yaml]: This implementation of per-kind-node bootid is actually per-agent-process bootid, which changes after agent restarts (upgrade). ci-ipsec-upgrade goes red due to that.

I'll add a hidden flag since everyone is okay with it. Hidden flag solution is more low-carbon (doesn't need to run a new ci build job) and it can be real per-kind-node bootid.

**Review by borkmann — APPROVED** [review]

**Review by tommyp1ckles — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]

**Review by ldelossa — APPROVED** [review]

**Comment by julianwiedmann** [comment]
@jschwinger233 this picked up a conflict and needs rebasing :/

**Comment by julianwiedmann** [comment]
> ci-ipsec-upgrade can't be covered in this PR because 1.16 still shares boot ID. Upgrade from 1.16 to 1.17 in CI will break connectivity as new cilium suddenly uses a different boot ID.

I added https://github.com/cilium/cilium/issues/36742 to keep track of this.

**Pull request merged**
